python/loan_application/app/implementation.py

```diff
@@ -312,6 +314,63 @@ def submit_confirmation(
         "merchant_payment_token": loan_application_id
     }
 
+'''
+When request is sent to API endpoint, submit_merchant_config checks that merchant
+is in our memory storage and if so updates: prequal, minimum_loan_amount and
+maximum_loan_amount with supplied info. This assumes that merchants not only can
+updated whether prequal is enabled but they can update their min and max loans if
+desired upon enabling prequal.
+'''
+def submit_merchant_config(
+        merchant_id: str,
+        body: dict
+) -> Union[Response, dict]:
+    #first we check if the merchant is in our database
+    merchant_to_configure = get_merchant_configuration(merchant_id)
+    if merchant_to_configure:
+        #use this info to instantiate a new class for this merchant with the updated values from API request
+        updated_configuration = MerchantConfiguration(
+            merchant_id=merchant_id,
+            name=merchant_to_configure.name,
+            **body
+        )
+        #store updated values for loan
+        minimum_loan_amount = updated_configuration.minimum_loan_amount
+        maximum_loan_amount = updated_configuration.maximum_loan_amount
+        prequal_enabled = updated_configuration.prequal_enabled
+        #check min and max loan amounts are valid
+        if maximum_loan_amount < minimum_loan_amount:
+            #if loan amounts are invalid, return failed response 400
+            content_type = mimetype = 'application/json'
+            return Response(
+                status=400, content_type=content_type, mimetype=mimetype,
+                response=json.dumps({
+                    "field": "minimum_loan_amount",
+                    "message": "Maximum loan amount must be larger than minimum loan amount."
+                })
+            )
+        #use updated values to update entry for merchant in database and retrieve configuration Id
+        merchant_configuration_id = update_merchant_prequal_configuration(
+            merchant_id,
+            minimum_loan_amount,
+            maximum_loan_amount,
+            prequal_enabled
+            )
+        #return sucessful response
+        if merchant_configuration_id:
+            return {
+                "merchant_configuration_id": merchant_configuration_id
+            }
+    #if merchant was not in our database, return failed response 400
```

Comment:
Small coding tip. Since this section is much shorter, I'd recommend putting it in this format.

```
if not merchant_to_configure:
    return Response (.... error)

# Other logic if it exists.
```

This is because after all that other code, it's hard to follow what this case is for.

Reply:
The way you have it earlier with the `if failure, return error` for the min/max is good.